### PR TITLE
feat: Add support for aes(128|256)gcm

### PR DESCRIPTION
**Pull request by flux-benj**
Close https://github.com/net-ssh/net-ssh/issues/834

### CODE REVIEW

File: lib/net/ssh/transport/gcm_cipher.rb [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+            #
+            def read_and_mac(data, mac, _sequence_number)
+              # The authentication tag will be placed in the MAC field at the end of the packet
+
+              # OpenSSL does not verify auth tag length
+              # GCM mode allows arbitrary sizes for the auth_tag up to 128 bytes and a single
+              #   byte allows authentication to pass. If single byte auth tags are possible
+              #   an attacker would require no more than 256 attempts to forge a valid tag.
+              #
+              raise 'incorrect auth_tag length' unless mac.to_s.length == mac_length
```
Comment (flux-benj) [inline lib/net/ssh/transport/gcm_cipher.rb:90]: Seeking opinions from maintainers on enforcing this.

**Comment by mfazekas** [comment]
@flux-benj thanks this looks and seems to work for simple smoke test

File: lib/net/ssh/transport/gcm_cipher.rb (hunk 1 above)
Comment (mfazekas) [reply lib/net/ssh/transport/gcm_cipher.rb:90]: Are you asking about making sure that auth_tag length is reasonable size? I think it's ok to add such check, later we can add option to lessen if needed, but I might misunderstand the question.

See also https://github.com/ruby/openssl/issues/63

**Review by mfazekas — APPROVED** [review]

File: lib/net/ssh/transport/gcm_cipher.rb (hunk 1 above)
Comment (flux-benj) [reply lib/net/ssh/transport/gcm_cipher.rb:90]: Yup that is what I was asking, just making sure maintainers were explicitly across it

**Comment by mfazekas** [comment]
@flux-benj amazing, thanks much!

**Comment by alexanderheckel** [comment]
Many thanks to everyone for their great work. Are there any plans to release a new version of RubyGems soon?

**Comment by mfazekas** [comment]
net-ssh 7.3.0.rc1 is now released pls test and report back

**Comment by alexanderheckel** [comment]
Sure, here we go ...

I tested three different server configurations:
- exclusively aes128-gcm@openssh.com
- exclusively aes256-gcm@openssh.com
- exclusively aes128-gcm@openssh.com and aes256-gcm@openssh.com

Using 7.3.0.rc1 all connections were successful \o/.

**Pull request merged**